.github/workflows/smoke-test.yaml

```diff
@@ -76,18 +75,28 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
 
-      - name: Build application
-        run: ${{ env.GRADLE_EXEC }} build
+      - name: Build Smoke Tests
+        run: ${{ env.GRADLE_EXEC }} buildSmokeTestsContainer -x test
 
-      - name: Run application in background, capture logs in app.log
-        run: |
-          ${{ env.GRADLE_EXEC }} :server:run 2> server/src/test/resources/app.log < /dev/null &
-          echo "Application started with PID $APP_PID"
-          sleep 10
+      - name: Set Project Version Env Variable
+        run: echo "GRADLE_PROJECT_VERSION=$(${{ env.GRADLE_EXEC }} -q printProjectVersion)" >> "$GITHUB_ENV"
 
-      - name: Print App Logs
-        run: cat server/src/test/resources/app.log
+      # TODO if I use 'docker compose' from a shell script (meaning I call a shell scrip here instead)
+      # I get permission denied and it exits (which makes sense). Locally with the tool act I am able
+      # to reproduce it as well. If I use 'docker-compose' from a shell script however, locally with
+      # act it seems to pass, so maybe will pass in gh actions as well, but when I pushed, it exited
+      # saying 'docker-compose not found'. In theory maybe we can have 'docker-compose' available
+      # in gh actions so we can have a single gradle script that will do all the job and we will
+      # not need all these steps that will use docker directly (which was the initial idea)
```

Comment:
Could we address that? See comment and past runs of actions. If I could address that so either the `docker` or `docker-compose` executables work, I could implement the initial idea of centralizing all the logic into a single `gradle` task.